pulsar-client-cpp/lib/stats/ConsumerStatsImpl.cc

```diff
@@ -21,57 +21,84 @@
 #include <lib/LogUtils.h>
 
 #include <functional>
+#include <sstream>
 
 namespace pulsar {
 DECLARE_LOG_OBJECT();
 
+struct ConsumerStatsImpl::AsyncCallbackLock
+{
+    std::mutex mutex;
+    bool enabled = true;
```

Comment:
Should it be an `atomic_bool`?

Reply:
Access to `enabled ` is always performed under mutex lock. Therefore atomic is not required here.